### ISSUE TITLE
Constructible Walls & Tables

### ISSUE THREAD

**Issue by TimFinucane** [RE-SS3D/SS3D#123]
This is an example-led task for implementing more of the tilemap system. The end result is:

As a player,
I want to be able to take an existing tile and construct a wall or table on it whilst playing the game,
So that i'm able to modify the map i'm on.

Completion of this task will require the following extensions to the tilemap system:

- Networking tilemap updates, so that all current players' maps are updated when a wall is constructed, and all players joining after construction see the wall
- An extensible way of updating the tilemap when you know what tile you want to change and what action you do to it, but nothing else.

This task should not have any construction stages - once it is clear the player intends to construct a wall or table, it gets immediately made, just to keep things simple.

Once this task is complete, the next step (in a separate task) can be to add stages of wall construction.

Creates Issues:

- Multi stage construction (requiring multiple tools and materials)
